projects/ngrx.io/content/guide/store/architecture.md

```diff
@@ -1 +1,418 @@
+
+Ok, so we have a reducer, so how do we use it? Well, that's just a matter of invoking the reducer, like so:
+
+<code-example header="invoking our reducer">
+let state = reducer('', { type: '[Language] change', payload: 'english' }); //english
+
+state = reducer(state, { type: '[Language] change', payload: 'french' })
+</code-example>
+
+As you can see above we are invoking the reducer first with an empty string and an action saying we want to change to `english`. At that point our `state` variable is updated to contain the value `english`. In our next invocation we change the value by passing the existing `state` variable as the first parameter and then as our second parameter we pass an action object that will change the value to `french`. 
+
+#### Introducing a store
+At this point we know about actions and reducers but we might not understand how the state makes it into a store, so what do we do? Well this is an implementation detail and it might differ depending on wether we are talking Redux for React or Angular or some other framework. However, let's look at a very small naive implementation so you get a sense of how it works:
+
+<code-example header="naive store">
+class Store {
+  constructor() {
+    this.state = {
+      language: '',
+      products: []
+    }
+  }
+
+  select(fn) {
+    return fn(this.state);
+  }
+
+  dispatch(action) {
+    this.state = compute(this.state, action);
+    // add a way to broadcast this store change to listeners
+  }
+
+  compute(state, action) {
+    return {
+      language: languageReducer(state.language, action),
+      products: productsReducer(state.products, action) 
+    }
+  }
+}
+</code-example>
+We have on purpose left out the part on how we subscribe to the store and how we broadcast a new store state after it has been changed by a call to `dispatch()`. The reason for that is that it's implementation specific to what Redux implementation we are talking about. In some implementations EventEmitters are being used, in NGRX it's RxJS. 
+
+At this point we have mentioned all major concepts in Redux like `action`, `reducer` and `store`. We've even shown a very naive implementation so you understand how all the pieces go together. Next is understanding where NGRX comes with its specific solution and especially why the heavy usage of Typescript makes sense and helps creating a better experience.
+
+## Introducing NGRX and the need for type safety
+NGRX is a Redux implementation meant for Angular and it makes heave use of Typescript and RxJS. The biggest reason is probably typesafety. 
+
+So why would we want to make them typesafe in the first place? Well, remember when we constructed an action in our previous section and assigned a string value to the `type` field? That is actually quite error prone, cause you might mispel that string value. It's maybe hard to understand why a mispel would be such a big deal but let's very briefly mention how actions are used. 
+
+### Dispatching an action
```

Comment:
~~Should it read better if this would be before `Introducing NGRX and the need for type safety`?~~